api/src/main/java/com/lightbend/lagom/javadsl/api/CircuitBreaker.java

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright (C) 2016 Lightbend Inc. <http://www.lightbend.com>
+ */
+package com.lightbend.lagom.javadsl.api;
+
+/**
+ * Circuit breaker descriptor.
+ *
+ * This describes how circuit breaking should be applied to a particular {@link ServiceCall}.
+ */
+public abstract class CircuitBreaker {
+
+    private CircuitBreaker() {}
+
+    /**
+     * Do not use a Circuit breaker for this service call.
+     */
+    public static CircuitBreaker none() {
+        return NamedCircuitBreaker.NONE;
+    }
+
+    /**
+     * Use a circuit breaker per node, if supported.
+     *
+     * Service locators that support per node circuit breaking will typically also remove the node from their routing
+     * pool while the circuit breaker is open, and bring it in for one call when it's half open.
+     *
+     * If not supported by the service locator, a circuit breaker per service may be used.
+     */
+    public static CircuitBreaker perNode() {
+        return NamedCircuitBreaker.PER_NODE;
+    }
+
+    /**
+     * Use a circuit breaker per service.
+     */
+    public static CircuitBreaker perService() {
+        return NamedCircuitBreaker.PER_SERVICE;
+    }
+
+    /**
+     * Use a circuit breaker identified by the given circuit breaker ID.
+     *
+     * @param id The ID of the circuit breaker.
+     * @return The circuit breaker descriptor.
+     */
+    public static CircuitBreaker identifiedBy(String id) {
+        return new CircuitBreakerId(id);
+    }
+
+    /**
+     * A named circuit breaker.
+     */
+    public static final class NamedCircuitBreaker extends CircuitBreaker {
```

Comment:
Any reason for not making both `NamedCircuitBreaker` and `CircuitBreakerId` private classes? (sorry if I've missed something)